### ISSUE TITLE
Boolean fields not behaving as expected in equals and hash tests

### ISSUE THREAD

**Issue by toaster91** [danhagberg/pojot#7]
Seeing the exception below for a simple Pojo with a 1 String and 1 Boolean field.
java.lang.AssertionError: [Equals method test failed for TenantOptin.  Non-equal values in value did not cause inequality for instances.] expected [true] but found [false]

The class itself is simple:

```
public class TenantOptin  {
  
  private String name = null;
  private Boolean value = null;

  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

  public Boolean getValue() {
    public Boolean getValue() {
    return value;
  }
  public void setValue(Boolean value) {
    this.value = value;
  }

 @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    TenantOptin tenantOptin = (TenantOptin) o;
    return Objects.equals(name, tenantOptin.name) &&
        Objects.equals(value, tenantOptin.value);
  }
 @Override
  public int hashCode() {
    return Objects.hash(name, value);
  }
}


```

**Comment by toaster91** [comment]
The hash testing error is:
java.lang.AssertionError: [Hashcode method test failed for TenantOptin.  Non-equal values in value did not cause different hashcode for instances., Equals method test failed for TenantOptin.  Non-equal values in value did not cause inequality for instances.] expected [true] but found [false]

**Comment by danhagberg** [comment]
Problem due to using the same value for both objects in the equals and hashCode testing.  This was due to the limited (2) values for boolean.  
Test added to reproduce problem.
Problem resolved. 
Fix released in version 1.0.1
